### ISSUE TITLE
Feat : Implementing Auth functionality. GSSOC'24

### ISSUE THREAD

**Issue by mukilan2815** [oxiton-foundation/click-metrics#149]
Description
When I click logout, it shows that authentication is not present. I will resolve this issue and also add a Google Sign-In feature.

Changes
Implement the Google Sign-In feature and fix the logout issue.

### Screenshots

![image](https://github.com/oxiton-foundation/click-metrics/assets/71129858/2e36eb82-9429-49c3-9740-b74586a646a3)

### Checklist

Before submitting your pull request, ensure that you have completed the following tasks:

- [x ] Gone through the [rules and guidelines](https://github.com/oxiton-foundation/click-metrics/issues/24) of the project.
- [x ] When you are creating PR, your PR should title follows the specified format: [Feature/Enhancement/Bug Fix]: Brief description of the change.
- [ x] Attached OSSI or GSSoC in the heading of the Issue.
- [ x] I have checked this issue is not already covered in another issue.

Expected Time to Complete
I estimate that it will take approximately 1 day to complete the changes proposed in this issue.

Additional Notes
This implementation includes:

Integration with Google Sign-In API.
Proper handling of authentication state during logout.
Ensuring the user experience is seamless and secure.

GSSOC'24

**Comment by Swarnendu0123** [comment]
We will be using firebase for authentication. But before that, we need to get the UI of the auth page. Please attach an UI of auth page.

**Comment by mukilan2815** [comment]
![image](https://github.com/oxiton-foundation/click-metrics/assets/71129858/777c9d09-c942-43e1-9a2d-4b8dc4c9ce2d)

if this is ok i can work on this. else you can suggest any ui i can replicate it.

**Comment by Swarnendu0123** [comment]
Use black button with white text instead of green. Otherwise it looks good to me. Use the existing UI lib that we are using.

**Comment by mukilan2815** [comment]
I need to know the logic. I have planned to make like - First user need to signin or signup to get access to home page or any other page. I think this is good too. What do you think?

**Comment by Swarnendu0123** [comment]
Yes that's it. We also need to redirect to the auth pages if the user is not logged in.

**Comment by Swarnendu0123** [comment]
But I want the UI for now as a pull request.

**Comment by Swarnendu0123** [comment]
Not expecting a big change in a single PR.

**Comment by mukilan2815** [comment]
Please check into  #151
